orion/primitives/tadgan.py

```diff
@@ -3,41 +3,61 @@
 import logging
 import math
 import tempfile
-from functools import partial
+from typing import Union, Optional
 
-import keras
 import numpy as np
 import pandas as pd
-from keras import backend as K
-from keras.layers import Input
-from keras.layers.merge import _Merge
-from keras.models import Model
-from mlprimitives.adapters.keras import build_layer
+import tensorflow as tf
 from mlprimitives.utils import import_object
+from numpy import ndarray
 from scipy import stats
+from tensorflow.keras import Model
+from tensorflow.keras.losses import MeanSquaredError
 
 from orion.primitives.timeseries_errors import reconstruction_errors
 
 LOGGER = logging.getLogger(__name__)
+tf.keras.backend.set_floatx('float64')
 
+LOSS_NAMES = [
+    ['cx_loss', 'cx_real', 'cx_fake', 'cx_gp'],
+    ['cz_loss', 'cz_real', 'cz_fake', 'cz_gp'],
+    ['eg_loss', 'eg_cx_fake', 'eg_cz_fake', 'eg_mse']
+]
 
-class RandomWeightedAverage(_Merge):
-    def _merge_function(self, inputs):
-        """
-        Args:
-            inputs[0] x     original input
-            inputs[1] x_    predicted input
-        """
-        alpha = K.random_uniform((64, 1, 1))
-        return (alpha * inputs[0]) + ((1 - alpha) * inputs[1])
 
+def build_layer(layer: dict, hyperparameters: dict):
+    layer_class = import_object(layer['class'])
+    layer_kwargs = layer['parameters'].copy()
+    # TODO: Upgrade to using tf.keras.layers.Wrapper in mlprimitives.
+    if issubclass(layer_class, tf.keras.layers.Wrapper):
```

Comment:
can you please open an [issue in MLPrimitives](https://github.com/MLBazaar/MLPrimitives/issues/) to mark this as an issue to update next release?